bip-0360.mediawiki

```diff
@@ -1,724 +1,452 @@
+initial stack element N,
 
-An increase in the witness discount must not be taken lightly. Parties may take advantage of this discount for purposes other than 
-authorizing transactions (e.g., storage of arbitrary data as seen with "inscriptions").
+tapleaf script,
 
-There was some hope of designing P2QRH such that discounted public keys and signatures could not be repurposed for the storage of 
-arbitrary data by requiring that they successfully be verified before being written to Bitcoin's blockchain, a.k.a. "JPEG resistance".
-Later research <ref>Bas Westerbaan (2025), [https://groups.google.com/g/bitcoindev/c/5Ff0jdQPofo jpeg resistance of various post-quantum signature schemes]</ref> 
-provided strong evidence that this was not a feasible approach for the NIST approved Post-Quantum signature algorithms.
-It is an open question if Post-Quantum signature algorithms can be designed to provide JPEG resistance.
+control block = [control byte, 32 * m byte Merkle path] # m is the depth of the Merkle tree
+</pre>
 
-==== Raising tapscript's stack element size ====
+The initial stack elements provide the same functionality as they do in P2TR. That is, they place elements on the stack to be evaluated by the script, a.k.a. the redeem script.
 
-A problem faced by any attempt to add PQ signatures to tapscript is that the stack elements in tapscript cannot be larger than 520 bytes
-because the MAX_SCRIPT_ELEMENT_SIZE=520. This is problematic because PQ signature algorithms often have signatures and 
-public keys in excess of 520 bytes. For instance:
+The control block is a 1 + 32 * m byte array, where the first byte is the control byte and the next 32*m bytes are the Merkle path to the script. The control byte is the same as the control byte in a P2TR control block, including the 7 bits which are used to specify the tapleaf version. The parity bit of the control byte is always 1, since P2TSH does not have a key-path spend. Unlike P2TR, we omit the public key from the control block as it is not needed in P2TSH. We maintain support for the optional annex in the witness (see Specification section below for more details)'''.'''
 
-* ML-DSA public keys are 1,312 bytes and signatures are 2,420 bytes
-* SLH-DSA public keys are 32 bytes and signatures are 7,856 bytes
+'''Rationale'''
 
-We will first look at our approach to the problem of PQ signatures and then give our solution for public keys larger than 520 bytes.
+Design of the P2TSH output type is guided by the following intentions:
 
-To keep P2QRH small and simple, we have opted not to raise the stack element size limit as part of P2QRH, but instead make this change when 
-adding PQ signatures. That said, we are not strongly opposed to putting this increase in P2QRH.
+1) Minimize changes to the network – we should reuse existing Bitcoin code and preserve existing software behavior, workflows, user expectations and compatibility whenever possible.
```

Comment:
We use mediawiki enumeration else where, why are use using "1)" here rather than "-"? Why make this change, it seems to only add inconsistency?